### DISCUSSION TITLE
How do I implement dependency dispose callback behavior in GetX?

### DISCUSSION THREAD

**Discussion by OliverRhyme** [jonataslaw/getx#1296] [Q&A]
I want to implement this kind of behavior on GetX where it calls a disposing function whenever it is removed.

I've tried `Get.find` on `onClose` but it disposes first the object before having a chance to call the `close()` method on the object.
```
void main() {
  runApp(
    Provider<MyDatabase>(
      create: (context) => MyDatabase(),
      child: MyFlutterApp(),
      dispose: (context, db) => db.close(),
   ),
  );
}
```

**Comment by eduardoflorence** [comment]
Show an example of how you tried to do with onClose (GetxController)

**Reply by OliverRhyme** [reply]
```
  @override
  void onClose() {
    Get.find<ASIDatabase>().close();
    super.onClose();
  }
```
This one on my controller which corresponds to the bindings where it is created
